plugins/uefi/fu-plugin-uefi.c

```diff
@@ -733,6 +743,13 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 	g_autoptr(GError) error_local = NULL;
 	g_autoptr(GPtrArray) entries = NULL;
 
+	/* vendor quirks */
+	str = fu_plugin_get_dmi_value (plugin, "BiosVendor");
+	if (g_strcmp0 (str, "LENOVO") == 0)
+		data->use_legacy_bootmgr_desc = TRUE;
+	if (g_strcmp0 (str, "American Megatrends Inc.") == 0)
+		data->use_shim_unique_filepath = TRUE;
```

Comment:
Not app AMI firmware suffers this problem as described.

Reply:
Right, but in the case of other OEMs (even like Lenovo) isn't the `BiosVendor` set to the OEM? Not hijacking the BiosVendor seems to be what smaller vendors do.

Reply:
It's incredibly difficult to hand a specification to a platform team if it will behave differently when they choose a different BIOS vendor. 

Without going into the details for economies of scale even with Dell there are certain "value" platforms that take a BIOS with minimal modifications from their IBV.

I really feel the right way to go about this is to be consistent in fwupd behavior and work with IBV to get their reference/core codebase to do the right thing. All the smaller guys who take it will get the right thing too then.

Reply:
> work with IBV to get their reference/core codebase to do the right thing

That's exactly what we're doing -- Peter Jones take on it is that what AMI is doing is technically allowed, but not in the spirit of the specification. Fixing the firmware also doesn't help the thousands of machines "in the wild" that are running the old firmware. When you reboot after deploying the new firmware update you're still using the "old" firmware and thus you never get the boot entry.

You could argue, that it would be a better default to *always* copy shim.efi on all platforms; shim is only ~1Mb and then we don't need a quirk at all.

Reply:
> Fixing the firmware also doesn't help the thousands of machines "in the wild" that are running the old firmware. When you reboot after deploying the new firmware update you're still using the "old" firmware and thus you never get the boot entry.

But this is a fixed quantity of machines that have a questionable de-duplication implementation.  Isn't it better to just quirk those machines specific SMBIOS data and the broken version combinations rather than apply this to "ALL" AMI?  What happens if there is some competing quirk needed for a swatch of machines later down the road?

> You could argue, that it would be a better default to always copy shim.efi on all platforms; shim is only ~1Mb and then we don't need a quirk at all.

I thought that the Lenovo case was trying to analyze the specific characteristics of the entry `Linux-Firmware-Updater\fwupdx64.efi`?  Doesn't that potentially break them too?

Reply:
> Isn't it better to just quirk those machines specific SMBIOS data

Yes, you're right. I think switching it around to use `Manufacturer=Star Labs` is probably much better. Ideally we could even do that from a quirk file, although quirk files are not really set up for random HwID matches like that right now.

> I thought that the Lenovo case was trying to analyze

I never got to the bottom of exactly what was happening. I assumed they were using the "label" only for the whitelist, but it's entirely possible they used something else.

Reply:
That looks to me like it will work, but I agree that quirking it with specific smbios data would be better. As for Lenovo, they've definitely been doing label matching for various things before, so it wouldn't surprise me if that's what they're doing here.

Reply:
I've just added something like this:

    # Star Labs LabTop (Mk III)
    [Guid=8265d473-a6c2-42b4-897b-bc220faa2d32]
    Flags = use-shim-unique-filepath

I'm reworking the patch to be a bit less invasive.

Reply:
Okay, patch updated to use the quirk.